test/Operators/logicalAndAssignment.js

```diff
@@ -0,0 +1,123 @@
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+assert.sameValue = function sameValue(expected, actual, message) {
+    assert.areEqual(actual, expected, message);
+}
+
+const tests = [
+    {
+        name: "lgcl and assignment operator",
+        body: function () {
+            var value = undefined;
+            assert.sameValue(value &&= 1, undefined, "(value &&= 1) === undefined; where value = undefined");
+
+            value = null;
+            assert.sameValue(value &&= 1, null, "(value &&= 1) === null where value = null");
+
+            value = false;
+            assert.sameValue(value &&= 1, false, "(value &&= 1) === false; where value = false");
+
+            value = 0;
+            assert.sameValue(value &&= 1, 0, "(value &&= 1) === 0; where value = 0");
+
+            value = -0;
+            assert.sameValue(value &&= 1, -0, "(value &&= 1) === -0; where value = -0");
+
+            value = NaN;
+            assert.sameValue(value &&= 1, NaN, "(value &&= 1) === NaN; where value = NaN");
+
+            value = "";
+            assert.sameValue(value &&= 1, "", '(value &&= 1) === "" where value = ""');
+
+
+
+            value = true;
+            assert.sameValue(value &&= 1, 1, "(value &&= 1) === 1; where value = true");
+
+            value = 2;
+            assert.sameValue(value &&= 1, 1, "(value &&= 1) === 1; where value = 2");
+
+            value = "test";
+            assert.sameValue(value &&= 1, 1, '(value &&= 1) === 1; where value = "test"');
+
+            var sym = Symbol("");
+            value = sym;
+            assert.sameValue(value &&= 1, 1, "(value &&= 1) === 1; where value = Symbol()");
+
+            var obj = {};
+            value = obj;
+            assert.sameValue(value &&= 1, 1, "(value &&= 1) === 1; where value = {}");
+        }
+    },
+    {
+        name: "lgcl and eval strict",
+        body: function () {
+            "use strict",
+                eval &&= 20;
+        }
+    },
+    {
+        name: "lgcl and arguments strict",
+        body: function () {
+            "use strict",
+                arguments &&= 20;
+        }
+    },
+    {
+        name: "lgcl and assignment lhs before rhs",
+        body: function () {
+            function DummyError() { }
+
+            class TestError extends Error {}
+
+            assert.throws(function () {
+                var base = null;
+                var prop = function () {
+                    throw new DummyError();
+                };
+                var expr = function () {
+                    throw new TestError("right-hand side expression evaluated");
+                };
+
+                base[prop()] &&= expr();
+            }, DummyError);
+
+            // assert.throws(function () {
+            //     var base = null;

```

Comment:
I assume you're planning more tests?

Perhaps have a look at test/es7/nullish.js for a model - tests I used for the `??` operator.

Also please put these in that folder. (es7 is used for new features that aren't closely tied to something existing)

Reply:
Yep. But something is wrong...I'm trying to find it out.

Reply:
The CI failures are due to the Trigraph issue I commented on above if you have other problems, let me know what's going wrong and I'll see if I can help.

If it compiles but doesn't run correctly for certain cases try putting the failing case in a file on its own, using a debug build of chakracore and running with the flag `-dump:bytecode` this will print out the bytecode instructions that are being generated so you can look at what exactly is happening - can also post that and I'll see if I can see the problem.